openpype/hosts/maya/plugins/publish/collect_arnold_scene_source.py

```diff
@@ -48,3 +44,13 @@ def process(self, instance):
             self.log.debug("No renderable cameras found.")
 
         self.log.debug("data: {}".format(instance.data))
+
+    def get_ass_data(self, objset, members, suffix):
+        if objset.endswith(suffix):
+            members = cmds.ls(members, long=True)
+            if not members:
+                return
+            children = get_all_children(members)
+            members = list(set(members) - set(children))
+
+            return children + members
```

Comment:
`get_ass_data` does not describe to me **what** it does.

Also, your logic above is invalid, it will now set `None` to one of the two entries since the object set will not match the suffix in that iteration, etc.

```suggestion
    def get_hierarchy(self, nodes):
        """Return nodes with all their children"""
        nodes = cmds.ls(nodes, long=True)
        if not nodes:
            return []
        children = get_all_children(nodes)

        # Make sure nodes merged with children only
        # contains unique entries
        return list(set(nodes) - set(children))

```